app/domain/authentication/authn_jwt/identity_provider_factory.rb

```diff
@@ -0,0 +1,23 @@
+module Authentication
+  module AuthnJwt
+    # Factory for jwt identity providers.
+    # If Identity variable is configured factory return the decoded_token_provider
+    # If the Identity variable is not configured and there is account field in url factory returns url provider
+    # If the above conditions are not met exception is raised
+    class IdentityProviderFactory
+      PROVIDERS = {
```

Comment:
i miss why do we need this dictionary ?

Reply:
Removed dictionary .  Created variables with this providers that i factory decides which to return